### PR TITLE
Generate .mta.json for Acheivements in pageJson output mode

### PR DESCRIPTION
**Pull request by mingwli**
[AB#394719](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/394719)

### CODE REVIEW

File: src/docfx/build/page/PageBuilder.cs [hunk 1]
```diff
@@ -173,16 +174,19 @@ public void Build(ErrorBuilder errors, FilePath file)
...
+                return (new JObject(), default);
             }
 
             var mime = _documentProvider.GetMime(file);
-
-            return (_templateEngine.RunJavaScript($"{mime}.json.js", sourceModel), new JObject());
+            var mtaJs = $"{mime}.mta.json.js";
+            var metadata = _config.OutputType == OutputType.PageJson && _templateEngine.JavaScriptExists(mtaJs)
+                ? _templateEngine.RunJavaScript(mtaJs, sourceModel) as JObject
+                : default;
```
Comment (live1206) [inline src/docfx/build/page/PageBuilder.cs:188]: could this be `new JObject()` to get rid of nullable?

File: src/docfx/build/page/PageBuilder.cs [hunk 2]
```diff
@@ -173,16 +174,19 @@ public void Build(ErrorBuilder errors, FilePath file)
             }
         }
 
-        private (object output, JObject metadata) CreateDataOutput(FilePath file, JObject sourceModel)
+        private (object output, JObject? metadata) CreateDataOutput(FilePath file, JObject sourceModel)
         {
             if (_config.DryRun)
             {
-                return (new JObject(), new JObject());
+                return (new JObject(), default);
```
Comment (live1206) [inline src/docfx/build/page/PageBuilder.cs:181]: get rid of nullable

File: src/docfx/template/TemplateEngine.cs [hunk 3]
```diff
@@ -113,6 +113,8 @@ public string RunMustache(ErrorBuilder errors, string templateName, JToken pageM
             return _mustacheTemplate.Render(errors, templateName, pageModel);
         }
 
+        public bool JavaScriptExists(string scriptName) => _package.Exists(new PathString($"ContentTemplate/{scriptName}"));
```
Comment (live1206) [inline src/docfx/template/TemplateEngine.cs:116]: This can be used in L121 as well

**Comment by mingwli** [comment]
image

File: src/docfx/build/page/PageBuilder.cs [hunk 4]
```diff
@@ -106,9 +106,10 @@ public void Build(ErrorBuilder errors, FilePath file)
                     _output.WriteJson(outputPath, output);
                 }
 
-                if (_config.OutputType == OutputType.PageJson && isContentRenderType)
+                if (_config.OutputType == OutputType.PageJson && metadata.Count > 0)
```
Comment (yufeih) [inline src/docfx/build/page/PageBuilder.cs:109]: revert this change, we only need to verify `.publish.json`

File: src/docfx/build/page/PageBuilder.cs [hunk 5]
```diff
@@ -182,7 +183,11 @@ public void Build(ErrorBuilder errors, FilePath file)
 
             var mime = _documentProvider.GetMime(file);
 
-            return (_templateEngine.RunJavaScript($"{mime}.json.js", sourceModel), new JObject());
+            return (
+                _templateEngine.RunJavaScript($"{mime}.json.js", sourceModel),
+                string.Equals("Achievements", mime, StringComparison.OrdinalIgnoreCase) // TODO: remove after schema exported
```
Comment (yufeih) [inline src/docfx/build/page/PageBuilder.cs:188]: consider an `if` check in line 186 for readability

**Review by yufeih — APPROVED** [review]
